src/connection_string.ts

```diff
@@ -136,6 +136,14 @@ export function resolveSRVRecord(options: MongoOptions, callback: Callback<HostA
           options.loadBalanced = true;
         }
 
+        if (
+          options.replicaSet &&
+          typeof options.srvMaxHosts === 'number' &&
```

Comment:
I think we should make the `srvMaxHosts` check consistent here and in the `validateLoadBalancedOptions` - i.e., either use "truthiness", or use the nonzero number check; I think at this point, the srvMaxHosts is always going to be a number, so we *can* use truthy for "non-zero", but we can also use `> 0` in both places to be more precise